acme/acme/client.py

```diff
@@ -527,21 +572,31 @@ def __del__(self):
         except Exception:  # pylint: disable=broad-except
             pass
 
-    def _wrap_in_jws(self, obj, nonce):
+    def _wrap_in_jws(self, obj, nonce, url):
         """Wrap `JSONDeSerializable` object in JWS.
 
         .. todo:: Implement ``acmePath``.
 
         :param .JSONDeSerializable obj:
+        :param str url: The URL to which this object will be POSTed
         :param bytes nonce:
         :rtype: `.JWS`
 
         """
         jobj = obj.json_dumps(indent=2).encode()
         logger.debug('JWS payload:\n%s', jobj)
-        return jws.JWS.sign(
-            payload=jobj, key=self.key, alg=self.alg,
-            nonce=nonce).json_dumps(indent=2)
+        kwargs = {
+            "alg": self.alg,
+            "nonce": nonce
+        }
+        if self.acme_version == 2:
+            # new ACME spec
+            kwargs["url"] = url
+            if self.account is not None:
```

Comment:
Is `account` only set for acmev2 uses? Is it required for acmev2 uses?

Reply:
It is needed for acmev2 uses. It is not needed otherwise.